libpkg/pkg_audit.c

```diff
@@ -542,25 +547,70 @@ pkg_audit_parse_vulnxml(struct pkg_audit *audit)
 {
 	XML_Parser parser;
 	struct vulnxml_userdata ud;
+	uid_t uid, euid;
+	pid_t pid;
+	struct passwd pwd, *res;
+	char pwd_buf[512];
 	int ret = EPKG_OK;
 
-	parser = XML_ParserCreate(NULL);
-	XML_SetElementHandler(parser, vulnxml_start_element, vulnxml_end_element);
-	XML_SetCharacterDataHandler(parser, vulnxml_handle_data);
-	XML_SetUserData(parser, &ud);
+	uid = getuid();
+	euid = geteuid();
+
+	if ((pid = fork()) == 0) {
```

Comment:
Assignment shouldn't be within the conditional, but outside of it.

Just a personal style note, for things like this, I like switch statements:

```
pid = fork();
switch (pid) {
case -1:
    do_the_error_thing();
    break;
case 0:
    do_the_child_thing();
    break;
default:
    do_the_parent_thing();
    break;
}
```